picard/ui/metadatabox.py

```diff
@@ -624,6 +624,8 @@ def _update_tags(self, new_selection=True, drop_album_caches=False):
 
             tag_diff.add('~length', str(orig_metadata.length), str(new_metadata.length),
                          removable=False, readonly=True)
+            if len(files) == 1 and (settings['rename_files'] or settings['move_files']):
+                tag_diff.add('~filepath', [file.filename], [file.make_filename(file.filename, orig_metadata)], removable=False, readonly=True)
```

Comment:
The new filename must be made from `new_metadata`, not `orig_metadata`.

Reply:
After testing this I think the path should also be shown in case the settings `rename_files` and `move_files` are disabled. Then it is just easier and both the old and new value are `file.filename`. I'd do something like this:

```python
            if len(files) == 1:
                if settings['rename_files'] or settings['move_files']:
                    new_filename = file.make_filename(file.filename, new_metadata)
                else:
                    new_filename = file.filename
                tag_diff.add('~filepath', [file.filename], [new_filename],
                             removable=False, readonly=True)
```